pkg/vmprovider/providers/vsphere/vmprovider_vm.go

```diff
@@ -659,6 +658,8 @@ func (vs *vSphereVMProvider) vmCreateGenConfigSpec(
 		createArgs.VMImageStatus.Firmware,
 		vmClassConfigSpec)
 
+	util.EnsureMinHardwareVersionInConfigSpec(createArgs.ConfigSpec, vmCtx.VM.Spec.MinHardwareVersion)
```

Comment:
We should probably move this after _the check for empty hardware version and PCI/PVC devices to set the min supported hw version._  in 673.. thoughts?. 

I see that it's handled right in the v1a2 implementation.

Reply:
Sure works either way, moving it after will ensure that the min h/w version for PCI devices is set, and if not needed, then the min version does get set.

Reply:
Yeah, Better to keep it after the PCI/PVCDevices block if the intent is to not override the block completely. 
Because 
- The PCI/PVC devices block will not get evaluated when the configSpec.hardwareVersion is set earlier by EnsureMinHardwareVersionInConfigSpec. 
- This also opens up the possibility for the VM's with PVC/PCI devices to have a hardware version set by EnsureMinHardwareVersionInConfigSpec that's lesser than 'min supported' for the device

Reply:
Makes sense, fixed